### PR TITLE
refactor(repositories): item like extends mutable repository

### PR DESCRIPTION
**Pull request by Defferrard**

### CODE REVIEW

File: src/services/item/plugins/itemLike/repository.ts [hunk 1]
```diff
@@ -1,96 +1,98 @@
...
-    if (!deleteResult.raw.length) {
-      throw new ItemLikeNotFound({ creatorId: creator.id, itemId: item.id });
+    if (!entity) {
+      throw new EntryNotFoundBeforeDeleteException(this.entity);
     }
 
-    return deleteResult.raw[0].id;
-  },
-});
+    super.delete(entity.id);
```
Comment (ReidyT) [inline src/services/item/plugins/itemLike/repository.ts:94]:
```suggestion
    await super.delete(entity.id);
```

**Review by ReidyT — APPROVED** [review]
Nice work 👻

File: src/services/item/plugins/itemLike/index.ts [hunk 2]
```diff
@@ -102,7 +102,7 @@ const plugin: FastifyPluginAsync = async (fastify) => {
           },
         };
         await actionService.postMany(member, buildRepositories(manager), request, [action]);
-        return newItemLike;
+        return newItemLike.id;
```
Comment (pyphilia) [inline src/services/item/plugins/itemLike/index.ts:105]: Need to reflect in schema?

**Review by pyphilia — APPROVED** [review]

File: src/services/item/plugins/itemLike/index.ts (hunk 2 above)
Comment (Defferrard) [reply src/services/item/plugins/itemLike/index.ts:105]: In fact, the schema is already valid. The previous implementation was not respecting the schema

**Review by spaenleh — APPROVED** [review]

**Pull request merged**